service/controller/resource/monitoring/prometheus/resource.go

```diff
@@ -141,6 +144,22 @@ func toPrometheus(ctx context.Context, v interface{}, config Config) (metav1.Obj
 
 	labels[key.MonitoringLabel] = "true"
 
+	nodeCount, err := countClusterNodes(ctx, config.K8sClient, cluster)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	prometheusMemory := key.PrometheusDefaultMemory()
+	if nodeCount > 2 {
```

Comment:
We set a minimum of 1Gi or RAM if node count is less than 2 nodes but then we multiply 500MB per node